src/components/interaction-designer/resource-editors/PerModeResourceEditor.vue

```diff
@@ -84,7 +85,38 @@ export class PerModeResourceEditor extends mixins(FlowUploader, Permissions, Rou
     [SupportedMode.OFFLINE, ['fas', 'mobile-alt']],
   ])
 
-  isPanelExpanded = {}
+  expandedPanels = {}
+  expandedPanelsSearch = {}
+
+  get hasOpenPanelsOverriden() {
+    return this.openResources === undefined || this.openResources.length > 0
+  }
+
+  get isPanelExpanded() {
+    return this.hasOpenPanelsOverriden
+      ? this.expandedPanelsSearch
+      : this.expandedPanels
+  }
+
+  set isPanelExpanded(value) {
+    if (this.hasOpenPanelsOverriden) {
+      this.expandedPanelsSearch = value
+    } else {
+      this.expandedPanels = value
+    }
+  }
```

Comment:
this setter might be removed if we modify the `updateIsPanelExpanded` a bit.
please see a proposal here:
https://github.com/FLOIP/flow-builder/commit/6e05d98c12187a42c47ede902ea89bf40f5b3e8c
(cherry-pick if you agree)
There are some types as well. And I would propose renaming expandedPanelsSearch -> expandedPanelsOverridden since we are agnostic of the search in this component and we are already using the 'overridden' term

Reply:
~~agnostic~~ unaware